app/models/spree/order_contents_decorator.rb

```diff
@@ -1,5 +1,26 @@
 module Spree
   OrderContents.class_eval do
+
+    def add_to_line_item(variant, quantity, options = {})
+      line_item = grab_line_item_by_variant(variant, false, options)
+
+      if line_item && part_variants_match?(line_item, variant, quantity, options)
+        line_item.quantity += quantity.to_i
+        line_item.currency = currency unless currency.nil?
+      else
+        opts = { currency: order.currency }.merge ActionController::Parameters.new(options).
```

Comment:
It probably would, but this is core. Not really wanting to deviate from what they have: https://github.com/spree/spree/blob/master/core/app/models/spree/order_contents.rb#L80